openedx/core/djangoapps/user_api/accounts/serializers.py

```diff
@@ -226,7 +228,10 @@ def to_representation(self, user):  # lint-amnesty, pylint: disable=arguments-di
             fields = _visible_fields(user_profile, user, self.configuration)
         else:
             fields = self.configuration.get('public_fields')
-
+        if isinstance(fields, list) and fields is not None:
+            fields.append('hide_username')
+        elif isinstance(fields, set) and fields is not None:
+            fields.add('hide_username')
```

Comment:
We shouldn't add this here like this. This should be handle like all the other fields through `ACCOUNT_VISIBILITY_CONFIGURATION["admin_fields"]`.

Reply:
Although i agree with Zainab's comment above^, but as i am also working on this code area i want to understand the reason why you had to check the type of fields if they set or list? Wont it be always a list? I am asking this so that i can avoid any bugs in my code.